src/Components/Patient/ShiftCreate.tsx

```diff
@@ -242,17 +235,22 @@ export const ShiftCreate = (props: patientShiftProps) => {
         ambulance_number: state.form.ambulance_number,
       };
 
-      const res = await dispatchAction(createShift(data));
-      setIsLoading(false);
+      await request(routes.createShift, {
+        body: data,
 
-      if (res && res.data && (res.status == 201 || res.status == 200)) {
-        await dispatch({ type: "set_form", form: initForm });
-        Notification.Success({
-          msg: "Shift request created successfully",
-        });
+        onResponse: ({ res, data }) => {
+          setIsLoading(false);
 
-        navigate(`/shifting/${res.data.id}`);
-      }
+          if (res && data && (res.status == 201 || res.status == 200)) {
```

Comment:
Simplify the expression. `.ok` already does these checks for you.

```suggestion
          if (res?.ok) {
```